store/generated_key/memstore/v2/memstore.go

```diff
@@ -0,0 +1,219 @@
+		return nil, err
+	}
+
+	// compute kzg data commitment. this is useful for testing
+	// READPREIMAGE functionality in the arbitrum x eigenda integration since
+	// preimage key is computed within the VM from hashing a recomputation of the data
+	// commitment
+	dataCommitment, err := verification.GenerateBlobCommitment(e.g1SRS, encodedVal)
+	if err != nil {
+		return nil, err
+	}
+
+	x := dataCommitment.X.BigInt(&big.Int{})
+	y := dataCommitment.Y.BigInt(&big.Int{})
+
+	g1CommitPoint := cert_verifier_binding.BN254G1Point{
+		X: x,
+		Y: y,
+	}
+
+	pseudoRandomBlobInclusionInfo := cert_verifier_binding.BlobInclusionInfo{
+		BlobCertificate: cert_verifier_binding.BlobCertificate{
+			BlobHeader: cert_verifier_binding.BlobHeaderV2{
+				Version:       0,                            // only supported version as of now
+				QuorumNumbers: []byte{byte(0x0), byte(0x1)}, // quorum 0 && quorum 1
+				Commitment: cert_verifier_binding.BlobCommitment{
+					LengthCommitment: cert_verifier_binding.BN254G2Point{
+						X: [2]*big.Int{randInt(1000), randInt(1000)},
+						Y: [2]*big.Int{randInt(1000), randInt(1000)},
+					},
+					LengthProof: cert_verifier_binding.BN254G2Point{
+						X: [2]*big.Int{randInt(1), randInt(1)},
+						Y: [2]*big.Int{randInt(1), randInt(1)},
+					},
+					Commitment: g1CommitPoint,
+					Length:     uint32(len(encodedVal)),
+				},
+				PaymentHeaderHash: [32]byte(randomBytes(32)),
+			},
+			Signature: randomBytes(48), // 384 bits
+			RelayKeys: []uint32{randUint32(), randUint32()},
+		},
+		BlobIndex:      uint32(randInt(1_000).Uint64()),
+		InclusionProof: randomBytes(128),
+	}
+
+	randomBatchHeader := cert_verifier_binding.BatchHeaderV2{
+		BatchRoot:            [32]byte(randomBytes(32)),
+		ReferenceBlockNumber: randUint32(),
+	}
+
+	randomNonSignerStakesAndSigs := cert_verifier_binding.NonSignerStakesAndSignature{
+		NonSignerQuorumBitmapIndices: []uint32{randUint32(), randUint32()},
+		NonSignerPubkeys: []cert_verifier_binding.BN254G1Point{
+			cert_verifier_binding.BN254G1Point{
+				X: randInt(1000),
+				Y: randInt(1000),
+			},
+		},
+		QuorumApks: []cert_verifier_binding.BN254G1Point{
+			cert_verifier_binding.BN254G1Point{
+				X: randInt(1000),
+				Y: randInt(1000),
+			},
+		},
+		ApkG2: cert_verifier_binding.BN254G2Point{
+			X: [2]*big.Int{randInt(1000), randInt(10000)},
+			Y: [2]*big.Int{randInt(1000), randInt(1000)},
+		},
+		QuorumApkIndices:  []uint32{randUint32(), randUint32()},
+		TotalStakeIndices: []uint32{randUint32(), randUint32(), randUint32()},
+		NonSignerStakeIndices: [][]uint32{
+			[]uint32{randUint32(), randUint32()},
+			[]uint32{randUint32(), randUint32()},
+		},
+		Sigma: cert_verifier_binding.BN254G1Point{
+			X: randInt(1000),
+			Y: randInt(1000),
+		},
+	}
+
+	artificialV2Cert := verification.EigenDACert{
+		BlobInclusionInfo:           pseudoRandomBlobInclusionInfo,
+		BatchHeader:                 randomBatchHeader,
+		NonSignerStakesAndSignature: randomNonSignerStakesAndSigs,
+	}
```

Comment:
can you put this in a function `generateRandomV2Cert()`

Reply:
[27bc9f6](https://github.com/Layr-Labs/eigenda-proxy/pull/305/commits/27bc9f6174c1498c61cc79b963ec52cb85b5ee88)